docs/web/contribute.md

```diff
@@ -43,88 +86,163 @@ assign the issue to yourself. This way, nobody else will work on it at the same
 issue that someone else is assigned to, please contact the assignee to know if they are still working
 on it.
 
+#### Make a branch
 
-## Working with backends
+**Dependencies:**
 
-Ibis comes with several backends. If you want to work with a specific backend, you will have to install
-the dependencies for the backend with `conda install -n ibis-dev -c conda-forge --file="ci/deps/<backend>.yml"`.
+- required: `git`
 
-If you don't have a database for the backend you want to work on, you can check the configuration of the
-continuos integration, where docker images are used for different backend. This is defined in
-`.github/workflows/main.yml`.
+Once you're inside of a nix-shell, the first thing you want to do is make a branch. Let's call it `useful-bugfix`.
+
+       :::sh
+       git checkout -b useful-bugfix
+
+#### Make the desired change
+
+**Dependencies:**
+
+- required: `git`
+
+Let's say you've made a change to `ibis/expr/types.py` to fix a bug reported in issue #9000 (a made up issue at the time of writing).
 
-## Run the test suite
+Running `git status` should give output similar to this:
 
-To run Ibis tests use the next command:
+        :::sh
+        $ git status
+        On branch useful-bugfix
+        Your branch is up to date with 'origin/useful-bugfix'.
+        
+        Changes not staged for commit:
+          (use "git add <file>..." to update what will be committed)
+          (use "git restore <file>..." to discard changes in working directory)
+                modified:   ibis/expr/types.py
+        
+        no changes added to commit (use "git add" and/or "git commit -a")
+
+#### Run the test suite
+
+**Dependencies:**
+
+- required: [`nix`](#nix) environment or [`conda`](#miniconda) environment
+
+To run Ibis tests use the following command:
 
 ```sh
 PYTEST_BACKENDS="sqlite pandas" python -m pytest ibis/tests
 ```
 
-You can change `sqlite pandas` by the backend or backends (space separated) that
-you want to test.
+You can change `sqlite pandas` to include one or more supported backends (space
+separated) that you want to test.
 
+It isn't necessary to provide `PYTEST_BACKENDS` at all, but it's useful for
+exercising more of the library.
 
-## Style and formatting
+#### Commit and push your changes
 
-We use [flake8](http://flake8.pycqa.org/en/latest/),
-[black](https://github.com/psf/black) and
-[isort](https://github.com/pre-commit/mirrors-isort) to ensure our code
-is formatted and linted properly. If you have properly set up your development
-environment by running ``make develop``, the pre-commit hooks should check
-that your proposed changes continue to conform to our style guide.
+**Dependencies:**
 
-We use [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) as
-our standard format for docstrings.
+- required: `git`
+- required: `cz` (available in the `nix` environment)
```

Comment:
is there a doc link on cz?

Reply:
I don't know what you're asking. Do you want me to put a link to the `cz` repository/documentation?